### DISCUSSION TITLE
Low performance - slow first SSR page request & slow pages tag (SEO) -  Total Blocking Time

### DISCUSSION THREAD

**Discussion by MarvingMoreton** [vercel/next.js#49758] [Help]
### Summary

I need help with optimizing the performance of a Next js Platform. I am struggling to make the pages making requests to a database not being "slow" and not being tagged as "slow pages" by SEO Audit tools.

Current pageSpeed test results:https://pagespeed.web.dev/analysis/https-dataaxy-com/cw7vbqm32r?form_factor=mobile

I have been making improvements for a month now, moving to a better database and all, it is slowly getting better but still not enough. The main remaining issue is the Total Blocking Time.

As I am a junior developer, maybe I am not using a built-in Next JS optimizer, like why is the main-thread work so long? (16s)
IS there a compiling/minifying/purging system to be manually activated with next js?

Thanks for the help!

### Additional information

```js
Issues pointed out by pageSpeed Insights now:
Largest Contentful Paint element Error!
Minimize main-thread work 16.2 s
Reduce JavaScript execution time 10.8 s

--> https://pagespeed.web.dev/analysis/https-dataaxy-com/zc7fjohyfd?form_factor=mobile
```

### Example

https://dataaxy.com/

**Comment by fuma-nama** [comment]
Database queries can be ***really*** slow when making requests in a serverless environment.

How do you make the request? 
Are you using SSR correctly? have you enabled caching? You can learn more about data fetching in the [Next.js Documentation](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching)

**Reply by MarvingMoreton** [reply]
Thank you for your reply @SonMooSans !

Indeed I am using SSR correctly. We are currently using the default caching system on the GET request/ fetch set up: https://nextjs.org/docs/app/building-your-application/data-fetching/caching

**Reply by fuma-nama** [reply]
> Thank you for your reply @SonMooSans !
> 
> 
> 
> Indeed I am using SSR correctly. We are currently using the default caching system on the GET request/ fetch set up: https://nextjs.org/docs/app/building-your-application/data-fetching/caching
> 
> 
> 
> 

It's hard to give you some suggestions without actually checking the code.

There're some of my suggestions:
- If the data fetched depends on user (such as session, user info), fetch it on client side
- Otherwise, use fetch with the `revalidate` property in a server component, notice that it's a millisecond value, you should not revalidate it too regularly

**Reply by fuma-nama** [reply]
Also, take extra care on which region your database and backend server in. If you're using Vercel, you can change the region in Settings.
